api/services/sandbox/v1/sandbox.proto

```diff
@@ -0,0 +1,132 @@
+/*
+	Copyright The containerd Authors.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+syntax = "proto3";
+
+// Sandbox is a v2 runtime extension that allows more complex execution environments for containers.
```

Comment:
I'm a little hesitating to position the sandbox API as an extension to the runtime/task API set or as another independent API set. We may also abstract them as two API set:
1) sandbox API: prepare environment to run pod/containers. The main functionality is to allocate resources and setup bundle.
2) runtime API: run pod/containers